Remove allowDummyToken functionality

passcod opened the pull request:
### Changes

- Follow up to #4150, removes `allowDummyToken`

### Checklist

- [x] Code is finished

passcod committed:
Remove allowDummyToken functionality

edmofro reviewed (approved):
Double approved! Handy at one point, but glad to see it go

passcod committed:
Remove obsolete scripts

These referenced fake-token, but they're also really really obsolete
(clone_server is for old sync, and migrate_mongo is for old db!)
